module/VuFind/tests/unit-tests/src/VuFindTest/Mailer/MailerTest.php

```diff
@@ -148,6 +148,52 @@ public function testSendWithFromOverride()
         $mailer->send('to@example.com', $address, 'subject', 'body');
     }
 
+    /**
+     * Test sending an email using an explicitly set reply-to address.
+     *
+     * @return void
+     */
+    public function testSendWithReplyTo()
+    {
+        $callback = function ($message) {
+            $fromString = $message->getFrom()->current()->toString();
+            return '<to@example.com>' == $message->getTo()->current()->toString()
+                && '<reply-to@example.com>' == $message->getReplyTo()->current()->toString()
+                && '<me@example.com>' == $fromString
+                && 'body' == $message->getBody()
+                && 'subject' == $message->getSubject();
+        };
+        $transport = $this->createMock(\Zend\Mail\Transport\TransportInterface::class);
+        $transport->expects($this->once())->method('send')->with($this->callback($callback));
+        $address = new Address('me@example.com');
+        $mailer = new Mailer($transport);
+        $mailer->send('to@example.com', $address, 'subject', 'body', null, 'reply-to@example.com');
+    }
+
+    /**
+     * Test sending an email using a from address override
+     * and an explicitly set reply-to address.
+     *
+     * @return void
+     */
+    public function testSendWithFromOverrideAndReplyTo()
+    {
+        $callback = function ($message) {
+            $fromString = $message->getFrom()->current()->toString();
+            return '<to@example.com>' == $message->getTo()->current()->toString()
+                && '<reply-to@example.com>' == $message->getReplyTo()->current()->toString()
+                && 'me <no-reply@example.com>' == $fromString
```

Comment:
Is it actually desired/expected that the text description on the from address should be "me" in this situation, or is that a bug discovered by this test? It looks to me like it might be an unexpected side effect of the way we override the from address. Let me know if you'd like me to investigate this more deeply -- it just looked strange at a glance.

Reply:
That's as designed. If you had a name with the address ("Me Myself <me@example.com>"), it would take the name and just replace the address. The idea was to keep it friendly-looking whenever possible. This can be revisited if you find it confusing.

Reply:
No, I think it's fine -- just wanted to be sure that was the intent.